torch/_inductor/sizevars.py

```diff
@@ -447,7 +447,21 @@ def strideof(name):
         # Assign all symbolic shapes needed to local variables
         needed = set(self.var_to_val.keys()) - set(self.replacements.keys())
 
-        for name, value in graph_inputs.items():
+        def is_expr(x):
+            return isinstance(x[1], sympy.Expr)
+
+        graph_inputs_expr = list(filter(is_expr, graph_inputs.items()))
+        graph_inputs_tensors = list(filter(lambda x: not is_expr(x), graph_inputs.items()))
+
+        for name, shape in graph_inputs_expr:
+            shape = self.simplify(shape)
```

Comment:
I'm a little confused here: is shape expected to be a single variable here? How do we know it always has to be a variable at this point?

Reply:
Maybe the answer is, needed only contains variables?

Reply:
Looks like it